### ISSUE TITLE
Seeding a new subsystem

### ISSUE THREAD

**Issue by gabrielcolson** [jgilbert01/aws-lambda-stream#394]
How would you seed a new subsystem with historical data ?

Example usecase:
- `Subsys X` has an egress-esg that emits external events and store them to its event lake
- we create `Subsys Y` and a routing rule to send external events sent by other subsytems to its event bus.
- but Subsys Y also needs historical data

Possible solution:
a script similar to the `aws-lambda-stream-cli` that reads from `Subsys X`'s event lake and sends the events to `Subsys Y`'s event bus. It could work but I fear that it would send duplicates to `Subsys Y`'s event lake.

Has anyone faced this problem before?

**Comment by vsnig** [comment]
I think the most straight forward solution is to replay events of interest to X's egress gateway which would make it publish external events that will seed the Y subsystem

**Comment by gabrielcolson** [comment]
That means you will get a lot of duplicated external events in both X and Y event lakes. Or should we not store external events to the event lakes ?

**Comment by orept** [comment]
No. I have `eventPattern: { source: ['custom'] }` rule for delivery stream that pushes events to event lake. That means that only internal events get pushed into the lake. But honestly even if they would get saved into the lake, I don't see a problem here (of course if it's not terrabytes of event you're trying to replay)

**Comment by vsnig** [comment]
No. I have eventPattern: { source: ['custom'] } rule for delivery stream that pushes events to event lake. That means that only internal events get pushed into the lake. But honestly even if they would get saved into the lake, I don't see a problem here (of course if it's not terrabytes of event you're trying to replay)

**Comment by vsnig** [comment]
No. I have `eventPattern: { source: ['custom'] }` rule for delivery stream that pushes events into event lake. That means that only internal events get pushed into the lake. But honestly even if they would get saved into the lake, I don't see a problem here (of course if it's not terrabytes of events you're trying to replay)

> Or should we not store external events to the event lakes ?

Haven't noticed the second question at first. I don't have a definite answer. Right now in John Gilbert's s3 [template](https://github.com/jgilbert01/templates/blob/e878611e258685d6630d13bc9a742b9225f89666/template-event-lake-s3/serverless/firehose.yml#L113) _anything-but-fault_ is saved. So it means that `external` events are saved too. In my lakes I have `source: ['custom']`, and I derived it from some previous version of template so it was like that at some point in the past. But logically, `external` events can't be replayed to anything within a subsystem so it seems there's little value of storing them in S3 lake, maybe just for statistcs sake.
Maybe John himself would want to say something on this topic.

**Comment by gabrielcolson** [comment]
Thanks for you insights.

In case you have multiple downstream subsystems, do you still replay events in the egress? We usually have an idempotency table in ingresses that expires records after 30 days to avoid sending duplicates. Should we keep those records forever to be able to replay as much as we want ?
